BetaCameras/ifm/O3D3xx.cs

```diff
@@ -381,22 +432,43 @@ protected override void ConnectImpl()
 
         private void UpdateLoop()
         {
+            int consecutiveFailCounter = 0;
+
             while (!_cancelUpdateThreadSource.Token.IsCancellationRequested)
             {
-                byte[] hdrBuffer = new byte[16];
-                Receive(_clientSocket, hdrBuffer, 0, 16, 300000);
-                var str = Encoding.Default.GetString(hdrBuffer, 5, 9);
-                Int32.TryParse(str, out int frameSize);
+                try
+                {
+                    byte[] hdrBuffer = new byte[16];
+                    Receive(_clientSocket, hdrBuffer, 0, 16, 300000);
+                    var str = Encoding.Default.GetString(hdrBuffer, 5, 9);
+                    Int32.TryParse(str, out int frameSize);
 
-                lock (_backLock)
+                    lock (_backLock)
+                    {
+                        if (_backBuffer.Length != frameSize)
+                        {
+                            _backBuffer = new byte[frameSize];
+                        }
+                        Receive(_clientSocket, _backBuffer, 0, frameSize, 300000);
+                        _frameAvailable.Set();
+                    }
+                }
+                catch(SocketException e)
                 {
-                    if (_backBuffer.Length != frameSize)
+                    consecutiveFailCounter++;
+                    log.Error($"{Name}: Socket Exception: {e.Message}");
+
+                    if(consecutiveFailCounter >= _maxConsecutiveReceiveFails)
                     {
-                        _backBuffer = new byte[frameSize];
+                        log.Error($"{Name}: Receive failt more than {_maxConsecutiveReceiveFails} times in a row. Shutting down update loop.");
```

Comment:
* "failed"

Perhaps: `$"{Name}: Receive failed {consecutiveFailCounter} times in a row. Shutting down update loop."`